src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -103,6 +103,7 @@ public long check(EC2Computer c) {
                 long currentTime = this.clock.millis();
 
                 if (currentTime > nextCheckAfter) {
+                    attemptReconnectIfOffline(c);
```

Comment:
Do we want to do this even if `DISABLED`, in which case AFAICT the other behaviors of the strategy are a no-op?